sql/expression/logic.go

```diff
@@ -47,6 +47,12 @@ func (a *And) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 	if err != nil {
 		return nil, err
```

Comment:
Probably should move the nil check on line 76 up to the top of the function and immediately return (nil, nil), rather than checking for lval and rval separately.